src/Base/Unit.cpp

```diff
@@ -86,371 +110,425 @@ Unit::Unit(int8_t Length, //NOLINT
+
+    sig.Value = sigVal("cbrt()",
+                       sig.Length                   / 3,
+                       sig.Mass                     / 3,
+                       sig.Time                     / 3,
+                       sig.ElectricCurrent          / 3,
+                       sig.ThermodynamicTemperature / 3,
+                       sig.AmountOfSubstance        / 3,
+                       sig.LuminousIntensity        / 3,
+                       sig.Angle                    / 3);
 
     Unit result;
-    result.Sig.Length                   = static_cast<int8_t>(Sig.Length                    * exp);
-    result.Sig.Mass                     = static_cast<int8_t>(Sig.Mass                      * exp);
-    result.Sig.Time                     = static_cast<int8_t>(Sig.Time                      * exp);
-    result.Sig.ElectricCurrent          = static_cast<int8_t>(Sig.ElectricCurrent           * exp);
-    result.Sig.ThermodynamicTemperature = static_cast<int8_t>(Sig.ThermodynamicTemperature  * exp);
-    result.Sig.AmountOfSubstance        = static_cast<int8_t>(Sig.AmountOfSubstance         * exp);
-    result.Sig.LuminousIntensity        = static_cast<int8_t>(Sig.LuminousIntensity         * exp);
-    result.Sig.Angle                    = static_cast<int8_t>(Sig.Angle                     * exp);
+    result.Val = sig.Value;
 
     return result;
 }
 
 bool Unit::isEmpty()const
 {
-    return (this->Sig.Length == 0)
-        && (this->Sig.Mass == 0)
-        && (this->Sig.Time == 0)
-        && (this->Sig.ElectricCurrent == 0)
-        && (this->Sig.ThermodynamicTemperature == 0)
-        && (this->Sig.AmountOfSubstance == 0)
-        && (this->Sig.LuminousIntensity == 0)
-        && (this->Sig.Angle == 0);
+    return Val == 0;
 }
 
-bool Unit::operator ==(const Unit& that) const
+int Unit::operator [](int index) const
 {
-    return (this->Sig.Length == that.Sig.Length)
-        && (this->Sig.Mass == that.Sig.Mass)
-        && (this->Sig.Time == that.Sig.Time)
-        && (this->Sig.ElectricCurrent == that.Sig.ElectricCurrent)
-        && (this->Sig.ThermodynamicTemperature == that.Sig.ThermodynamicTemperature)
-        && (this->Sig.AmountOfSubstance == that.Sig.AmountOfSubstance)
-        && (this->Sig.LuminousIntensity == that.Sig.LuminousIntensity)
-        && (this->Sig.Angle == that.Sig.Angle);
+    UnitSignature sig;
+    sig.Value = Val;
+
+    switch (index) {
+       case 0:
+          return sig.Length;
+       case 1:
+          return sig.Mass;
+       case 2:
+          return sig.Time;
+       case 3:
+          return sig.ElectricCurrent;
+       case 4:
+          return sig.ThermodynamicTemperature;
+       case 5:
+          return sig.AmountOfSubstance;
+       case 6:
+          return sig.LuminousIntensity;
+       case 7:
+          return sig.Angle;
+       default:
+          throw Base::RuntimeError("Unknown Unit element");
```

Comment:
Minor thing but here better an IndexError should be thrown.

Reply:
Fixed.